### PR TITLE
feat(consolidated-interstitial): add joinFlowVersion as CA event option

### PR DESCRIPTION
**Pull request by chburket**
# COMPLETES #https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-576382

## This pull request addresses

Added new joinFlowVersion CA event option that is needed for the consolidated interstitial feature.

https://sqbu-github.cisco.com/WebExSquared/event-dictionary/pull/1854

## by making the following changes

* bumped the "@webex/event-dictionary-ts" version
* added `MetricEventJoinFlowVersion` type
* pass through joinFlowVersion in `createClientEventObjectInMeeting`

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

Unit tests

### I certified that

- [X] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [X] I discussed changes with code owners prior to submitting this pull request

- [X] I have not skipped any automated checks
- [X] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new parameter `joinFlowVersion` for enhanced event tracking in call diagnostics.
	- Added a new type `MetricEventJoinFlowVersion` to improve event categorization.
	- Enhanced event submission capabilities with additional identifiers in the `submitClientEvent` method.

- **Bug Fixes**
	- Improved error handling and validation for client event submissions, ensuring more robust diagnostics.

- **Documentation**
	- Updated method signatures and interfaces to reflect new parameters and types for better clarity and usability.

### CODE REVIEW

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts [hunk 1]
```diff
@@ -2757,6 +2757,25 @@ describe('internal-plugin-metrics', () => {
...
+        const fetchOptions = await cd.buildClientEventFetchRequestOptions({
+          name: 'client.exit.app',
+          payload: {trigger: 'user-interaction', canProceed: false},
+          options,
+        });
+
+        assert
+          .equal(fetchOptions.body.metrics[0].eventPayload.event.joinFlowVersion)
+          .toBe(options.joinFlowVersion);
+      });
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts]: I think coderabbit is right here with the test

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts [hunk 2]
```diff
@@ -718,6 +719,10 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
       webexSubServiceType: this.getSubServiceType(meeting),
     };
 
+    if (joinFlowVersion) {
```
Comment (chrisadubois) [inline packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts]: doesn't this need to be added to the preMeeting function and all the other functions in this file? similar to sessionCorrelationId?

Comment (chburket) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts]: just preMeeting function, I think.  Good catch.  Added.

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (chburket) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts]: fixed

**Review by chrisadubois — APPROVED** [review]

File: packages/@webex/internal-plugin-metrics/src/metrics.types.ts [hunk 3]
```diff
@@ -110,6 +110,8 @@ export type MetricEventVerb =
   | 'warn'
   | 'exit';
 
+export type MetricEventJoinFlowVersion = 'Other' | 'NewFTE';
```
Comment (sreenara) [inline packages/@webex/internal-plugin-metrics/src/metrics.types.ts:113]: question: What does NewFTE indicate? I was unable to get this information from the media SPARK jira too.

**Review by sreenara — APPROVED** [review]
Approved with a single question

File: packages/@webex/internal-plugin-metrics/src/metrics.types.ts (hunk 3 above)
Comment (chburket) [reply packages/@webex/internal-plugin-metrics/src/metrics.types.ts:113]: New First Time Experience.  It just means the consolidated interstitial flow (where we skip straight to the interstitial page for guest users and allow them to enter their display name there).

Comment (chburket) [reply packages/@webex/internal-plugin-metrics/src/metrics.types.ts:113]: image

**Pull request merged**